tests/utils/test_file.py

```diff
@@ -39,31 +39,36 @@ def test_checking_file():
 @pytest.mark.utility
 def test_is_symlink():
     # Target path of the symbolic link
-    link_target = tempfile.mkdtemp()
+    link_target = tempfile.NamedTemporaryFile(dir=os.path.expanduser("~"))
 
     # Symbolic link path
-    link_path = os.path.join("$HOME", "test_dir_symlink")
+    link_path = tempfile.NamedTemporaryFile(dir=os.path.expanduser("~"))
+
+    link_path.close()
 
     # Create symbolic link
-    os.symlink(link_target, os.path.expandvars(link_path))
+    os.symlink(link_target.name, link_path.name)
+
+    # get filename from the path
+    filename = os.path.split(link_path.name)[-1]
```

Comment:
i think this should work but it is a bit cryptic using arrays. I think this is equivalent to running [os.path.basename](https://docs.python.org/3/library/os.path.html#os.path.basename) you can do `os.path.basename(link_path.name)`. I think these two statements are equivalent but a bit more safer.